Best Practices/Challenges/UnitTests/UnitTests.Challenge/NotificationWorker.cs

```diff
@@ -0,0 +1,65 @@
+﻿using UnitTests.Challenge.Interfaces;
+using UnitTests.Challenge.Interfaces.NotificationServices;
+
+namespace UnitTests.Challenge
+{
+    /// <summary>
+    /// This part involves writing tests for part 6, follow the document for more information
+    /// </summary>
+    public class NotificationWorker
+    {
+        private const int MaxRetryAttempts = 3;
+        private const int RetryDelayMilliseconds = 1;
+
+        private readonly INotificationService _notificationService;
+
+        public NotificationWorker(INotificationService notificationService)
+        {
+            _notificationService = notificationService;
+        }
+
+        public async Task SendNotification(string message)
```

Comment:
issue (blocking): We need to rethink this method and the test tasks with it. Here are my thoughts, but let's catchup in person:

- We can drop the IFaxNotificationService (and other) abstractions, as the method depends on details.
- We should add different methods to the INotificationService for the types of order status and switch on the order status (Dispatched, Cancelled, Delivered)